lib/krikri/software_agent.rb

```diff
@@ -30,6 +30,38 @@ def log(priority, msg)
     end
 
     ##
+    # Store this agent's generator activity, which is the activity that
+    # produced the target entities upon which the current agent will operate.
+    #
+    # It is assumed that the class that includes SoftwareAgent will define
+    # class methods .entity_behavior and .generator_entity_behavior, which
+    # return the class of the appropriate behavior.
+    #
+    # We look for either `generator_activity' or `generator_uri' in the `opts'
+    # keys.  `generator_activity' must be a Krikri::Activity object, and
+    # `generator_uri' can be a string or RDF::URI
+    #
+    # @see Krikri::Mapper::Agent
+    # @see Krikri::Harvester
+    #
+    def set_generator_activity!(opts)
```

Comment:
Can we just clean this up, choose an option name/format and move forward.  This is a lot of code/validation for something that seems like ti could be really simple.